lighthouse-core/gather/driver.js

```diff
@@ -247,7 +250,13 @@ class Driver {
         includeCommandLineAPI: true,
         awaitPromise: true,
         returnByValue: true,
-      }).then(result => {
+      };
+
+      if (useIsolation && typeof this._isolatedExecutionContextId === 'number') {
```

Comment:
if isolation is requested but there is no contextId... should we throw?

Reply:
I had this originally to check for leaks and it seemed ok, but it's definitely possible and seemed bad to tank the run because it since it won't matter most of the time

perhaps something to add to sentry though when it lands?

Reply:
Seems like we should be able to figure out definitively if it's possible to reach here without having created a new context? Then future violations of this can complain loudly by throwing (alternatively, see other comment about lazily calling `_createIsolatedWorld()`)

Reply:
going the lazy route 👍